src/gp_activetable.c

```diff
@@ -447,13 +447,8 @@ diskquota_fetch_table_stat(PG_FUNCTION_ARGS)
 	{
 		MemoryContext oldcontext;
 		TupleDesc     tupdesc;
-		int           ret_code = SPI_connect();
-		if (ret_code != SPI_OK_CONNECT)
-		{
-			ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR),
-			                errmsg("unable to connect to execute internal query. return code: %d.", ret_code)));
-		}
-		SPI_finish();
+
+		SPI_finish_if(SPI_connect_if_not_yet());
```

Comment:
Why is SPI_finish_if called right after SPI_connect_if_not_yet?

Reply:
That's how it was before the patch! I think it's some kind of test.

Reply:
So may be don't change this code, because SPI_connect and SPI_finish are not the same as SPI_connect_if_not_yet and SPI_finish_if?

Reply:
may be, we should remove [this](https://github.com/arenadata/diskquota/commit/e3edb27b589125a0e12394ffb5e364b8e7521026#diff-f49a88dcbab4d58987d955e5c9b273700b9678abce582c1d2bb456aba3f92c89R368-R374) dead code?

Reply:
Yes. I think SPI_connect and SPI_finish made sence because of get_ext_major_version, but now they are redundant

Reply:
removed